pkg/k8shandler/prometheus_rule.go

```diff
@@ -21,27 +24,42 @@ const (
 )
 
 func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdatePrometheusRules() error {
-
+	ctx := context.Background()
 	dpl := elasticsearchRequest.cluster
 
-	ruleName := fmt.Sprintf("%s-%s", dpl.Name, "prometheus-rules")
+	name := fmt.Sprintf("%s-%s", dpl.Name, "prometheus-rules")
 	owner := getOwnerRef(dpl)
 
-	promRule, err := buildPrometheusRule(ruleName, dpl.Namespace, dpl.Labels)
+	rule, err := buildPrometheusRule(name, dpl.Namespace, dpl.Labels)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to build prometheus rule: %w", err)
 	}
 
-	addOwnerRefToObject(promRule, owner)
+	addOwnerRefToObject(rule, owner)
 
-	err = elasticsearchRequest.client.Create(context.TODO(), promRule)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return err
+	err = elasticsearchRequest.client.Create(ctx, rule)
+	if err == nil {
+		return nil
+	}
+	if !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create prometheus rule: %w", err)
 	}
 
-	//TODO: handle update - use retry.RetryOnConflict
+	current := &monitoringv1.PrometheusRule{}
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		err = elasticsearchRequest.client.Get(context.TODO(), types.NamespacedName{Name: rule.Name, Namespace: rule.Namespace}, current)
+		if err != nil {
+			logrus.Debugf("could not get prometheus rule %q: %v", rule.Name, err)
```

Comment:
`%w` ?

Reply:
This actually has to be `%v` because `%w` is for fmt.Errorf to wrap errors. This is just a log so it doesn't apply.